diff --git a/src/import/qtquickvirtualkeyboardplugin.cpp b/src/import/qtquickvirtualkeyboardplugin.cpp
index 969884c277..9cfa0fa30b 100644
--- a/src/import/qtquickvirtualkeyboardplugin.cpp
+++ b/src/import/qtquickvirtualkeyboardplugin.cpp
@@ -142,6 +142,7 @@ void QtQuickVirtualKeyboardPlugin::registerTypes(const char *uri)
     qmlRegisterType(QUrl(componentsPath + QLatin1String("TraceInputKey.qml")), uri, 2, 0, "TraceInputKey");
     qmlRegisterType(QUrl(componentsPath + QLatin1String("WordCandidatePopupList.qml")), uri, 2, 0, "WordCandidatePopupList");
     qmlRegisterType(QUrl(componentsPath + QLatin1String("PopupList.qml")), uri, 2, 3, "PopupList");
+    qmlRegisterType(QUrl(componentsPath + QLatin1String("Preloader.qml")), uri, 1, 0, "Preloader");
     qmlRegisterType(QUrl(componentsPath + QLatin1String("SelectionControl.qml")), uri, 2, 1, "SelectionControl");
     qmlRegisterType(QUrl(componentsPath + QLatin1String("InputModeKey.qml")), uri, 2, 3, "InputModeKey");
 
diff --git a/src/virtualkeyboard/content/components/Keyboard.qml b/src/virtualkeyboard/content/components/Keyboard.qml
index da0648209d..30a38285e7 100644
--- a/src/virtualkeyboard/content/components/Keyboard.qml
+++ b/src/virtualkeyboard/content/components/Keyboard.qml
@@ -57,6 +57,7 @@ Item {
     readonly property bool formattedNumbersOnly: InputContext.inputMethodHints & Qt.ImhFormattedNumbersOnly
     readonly property bool digitsOnly: InputContext.inputMethodHints & Qt.ImhDigitsOnly
     property string layout
+    property string symbolLayout
     property string layoutType: {
         if (keyboard.handwritingMode) return "handwriting"
         if (keyboard.dialableCharactersOnly) return "dialpad"
@@ -805,7 +806,7 @@ Item {
             LayoutMirroring.enabled: false
             LayoutMirroring.childrenInherit: true
 
-            Loader {
+            Preloader {
                 id: keyboardLayoutLoader
                 objectName: "keyboardLayoutLoader"
 
@@ -822,6 +823,13 @@ Item {
                     when: keyboard.layout.length > 0
                 }
 
+                Binding {
+                    target: keyboardLayoutLoader
+                    property: "preloadSource"
+                    value: keyboard.symbolLayout
+                    when: keyboard.symbolLayout.length > 0
+                }
+
                 onItemChanged: {
                     // Reset input mode if the new layout wants to override it
                     if (item && item.inputMode !== -1)
@@ -892,7 +900,8 @@ Item {
                         }
                     }
                     function keyOnPoint(px, py) {
-                        var parentItem = keyboardLayoutLoader
+                        var parentItem = keyboardLayoutLoader.item
+                        if (!parentItem) return null;
                         var child = parentItem.childAt(px, py)
                         while (child !== null) {
                             var position = parentItem.mapToItem(child, px, py)
@@ -1410,6 +1419,7 @@ Item {
             newLayout = findLayout(locale, "main")
         }
         layout = newLayout
+        symbolLayout = findLayout(locale, "symbols")
         inputLocale = locale
         updateInputMethod()
     }
diff --git a/src/virtualkeyboard/content/components/Preloader.qml b/src/virtualkeyboard/content/components/Preloader.qml
new file mode 100644
index 0000000000..42aab68fbf
--- /dev/null
+++ b/src/virtualkeyboard/content/components/Preloader.qml
@@ -0,0 +1,31 @@
+/****************************************************************************
+**
+** Copyright (C) 2022 UltiMaker
+**
+****************************************************************************/
+
+import QtQuick 2.0
+
+Item {
+    property string preloadSource: ""
+    property string source: ""
+    readonly property QtObject item : (preloadSource === source)? preloaded.item : other.item
+    readonly property int status : (preloadSource === source)? preloaded.status : other.status
+    Loader {
+        id: preloaded
+        anchors.fill: parent
+        visible: parent.preloadSource === parent.source
+    }
+    Loader {
+        id: other
+        anchors.fill: parent
+        visible: parent.preloadSource !== parent.source
+    }
+    onPreloadSourceChanged: {
+        if (preloadSource.length > 0) preloaded.source = preloadSource;
+        if (source.length > 0 && preloadSource !== source) other.source = source;
+    }
+    onSourceChanged: {
+        if (source.length > 0 && preloadSource !== source) other.source = source;
+    }
+}
diff --git a/src/virtualkeyboard/content/layouts/fallback/symbols.qml b/src/virtualkeyboard/content/layouts/fallback/symbols.qml
index 26d629515e..f31c7a4553 100644
--- a/src/virtualkeyboard/content/layouts/fallback/symbols.qml
+++ b/src/virtualkeyboard/content/layouts/fallback/symbols.qml
@@ -31,15 +31,11 @@ import QtQuick 2.0
 import QtQuick.Layouts 1.0
 import QtQuick.VirtualKeyboard 2.1
 
-KeyboardLayoutLoader {
-    property bool secondPage
-    onVisibleChanged: if (!visible) secondPage = false
-    sourceComponent: secondPage ? page2 : page1
-    Component {
-        id: page1
-        KeyboardLayout {
+KeyboardLayout {
+            property bool secondPage
             keyWeight: 160
             KeyboardRow {
+                visible: !secondPage
                 Key {
                     key: Qt.Key_1
                     text: "1"
@@ -83,6 +79,7 @@ KeyboardLayoutLoader {
                 BackspaceKey {}
             }
             KeyboardRow {
+                visible: !secondPage
                 FillerKey {
                     weight: 56
                 }
@@ -127,6 +124,7 @@ KeyboardLayoutLoader {
                 }
             }
             KeyboardRow {
+                visible: !secondPage
                 keyWeight: 156
                 Key {
                     displayText: "1/2"
@@ -177,6 +175,7 @@ KeyboardLayoutLoader {
                 }
             }
             KeyboardRow {
+                visible: !secondPage
                 keyWeight: 154
                 SymbolModeKey {
                     weight: 217
@@ -202,13 +201,8 @@ KeyboardLayoutLoader {
                     weight: 204
                 }
             }
-        }
-    }
-    Component {
-        id: page2
-        KeyboardLayout {
-            keyWeight: 160
             KeyboardRow {
+                visible: secondPage
                 Key {
                     key: Qt.Key_AsciiTilde
                     text: "~"
@@ -253,6 +247,7 @@ KeyboardLayoutLoader {
                 BackspaceKey {}
             }
             KeyboardRow {
+                visible: secondPage
                 FillerKey {
                     weight: 56
                 }
@@ -297,6 +292,7 @@ KeyboardLayoutLoader {
                 }
             }
             KeyboardRow {
+                visible: secondPage
                 keyWeight: 156
                 Key {
                     displayText: "2/2"
@@ -347,6 +343,7 @@ KeyboardLayoutLoader {
                 }
             }
             KeyboardRow {
+                visible: secondPage
                 keyWeight: 154
                 SymbolModeKey {
                     weight: 217
@@ -371,6 +368,4 @@ KeyboardLayoutLoader {
                     weight: 204
                 }
             }
-        }
-    }
 }
diff --git a/src/virtualkeyboard/content/virtualkeyboard_content.qrc b/src/virtualkeyboard/content/virtualkeyboard_content.qrc
index 1e6392f44a..1526c10edc 100644
--- a/src/virtualkeyboard/content/virtualkeyboard_content.qrc
+++ b/src/virtualkeyboard/content/virtualkeyboard_content.qrc
@@ -28,6 +28,7 @@
         <file>components/HandwritingModeKey.qml</file>
         <file>components/WordCandidatePopupList.qml</file>
         <file>components/PopupList.qml</file>
+        <file>components/Preloader.qml</file>
         <file>components/SelectionControl.qml</file>
         <file>components/ShadowInputControl.qml</file>
         <file>components/InputModeKey.qml</file>
