diff --git a/debugfs/debugfs.c b/debugfs/debugfs.c
index 72b5cbb..165f924 100644
--- a/debugfs/debugfs.c
+++ b/debugfs/debugfs.c
@@ -26,9 +26,6 @@ extern char *optarg;
 #include <errno.h>
 #endif
 #include <fcntl.h>
-#ifdef HAVE_SYS_SYSMACROS_H
-#include <sys/sysmacros.h>
-#endif
 
 #include "debugfs.h"
 #include "uuid/uuid.h"
diff --git a/lib/blkid/devname.c b/lib/blkid/devname.c
index 671e781..3e2efa9 100644
--- a/lib/blkid/devname.c
+++ b/lib/blkid/devname.c
@@ -36,9 +36,6 @@
 #if HAVE_SYS_MKDEV_H
 #include <sys/mkdev.h>
 #endif
-#ifdef HAVE_SYS_SYSMACROS_H
-#include <sys/sysmacros.h>
-#endif
 #include <time.h>
 
 #include "blkidP.h"
diff --git a/lib/blkid/devno.c b/lib/blkid/devno.c
index 480030f..aa6eb90 100644
--- a/lib/blkid/devno.c
+++ b/lib/blkid/devno.c
@@ -31,9 +31,6 @@
 #if HAVE_SYS_MKDEV_H
 #include <sys/mkdev.h>
 #endif
-#ifdef HAVE_SYS_SYSMACROS_H
-#include <sys/sysmacros.h>
-#endif
 
 #include "blkidP.h"
 
diff --git a/lib/ext2fs/finddev.c b/lib/ext2fs/finddev.c
index 62fa0db..311608d 100644
--- a/lib/ext2fs/finddev.c
+++ b/lib/ext2fs/finddev.c
@@ -31,9 +31,6 @@
 #if HAVE_SYS_MKDEV_H
 #include <sys/mkdev.h>
 #endif
-#ifdef HAVE_SYS_SYSMACROS_H
-#include <sys/sysmacros.h>
-#endif
 
 #include "ext2_fs.h"
 #include "ext2fs.h"
diff --git a/lib/ext2fs/ismounted.c b/lib/ext2fs/ismounted.c
index 7d52471..bcac0f1 100644
--- a/lib/ext2fs/ismounted.c
+++ b/lib/ext2fs/ismounted.c
@@ -49,9 +49,6 @@
 #if HAVE_SYS_TYPES_H
 #include <sys/types.h>
 #endif
-#ifdef HAVE_SYS_SYSMACROS_H
-#include <sys/sysmacros.h>
-#endif
 
 #include "ext2_fs.h"
 #include "ext2fs.h"
diff --git a/misc/create_inode.c b/misc/create_inode.c
index 8ce3faf..ae22ff6 100644
--- a/misc/create_inode.c
+++ b/misc/create_inode.c
@@ -22,10 +22,6 @@
 #include <attr/xattr.h>
 #endif
 #include <sys/ioctl.h>
-#ifdef HAVE_SYS_SYSMACROS_H
-#include <sys/sysmacros.h>
-#endif
-
 #include <ext2fs/ext2fs.h>
 #include <ext2fs/ext2_types.h>
 #include <ext2fs/fiemap.h>
diff --git a/misc/mk_hugefiles.c b/misc/mk_hugefiles.c
index 00e95cd..71a15c5 100644
--- a/misc/mk_hugefiles.c
+++ b/misc/mk_hugefiles.c
@@ -35,9 +35,6 @@ extern int optind;
 #include <sys/ioctl.h>
 #include <sys/types.h>
 #include <sys/stat.h>
-#ifdef HAVE_SYS_SYSMACROS_H
-#include <sys/sysmacros.h>
-#endif
 #include <libgen.h>
 #include <limits.h>
 #include <blkid/blkid.h>
