diff -purN connman-1.32/src/service.c connman-1.32.orig/src/service.c
--- connman-1.32/src/service.c	2016-05-04 14:08:26.559059101 -0400
+++ connman-1.32.orig/src/service.c	2016-05-04 14:08:06.811059970 -0400
@@ -2863,7 +2863,7 @@ int __connman_service_check_passphrase(e
 				if (!isxdigit((unsigned char)
 					      passphrase[i]))
 					return -ENOKEY;
-		} else if (length > 63)
+		} else if (length < 8 || length > 63)
 			return -ENOKEY;
 		break;
 	case CONNMAN_SERVICE_SECURITY_WEP:
