--- a/src/technology.c    2015-04-19 18:16:21.000000000 -0400
+++ b/src/technology.c    2016-01-25 15:06:55.617586327 -0500
@@ -923,7 +923,7 @@
 		if (technology->type != CONNMAN_SERVICE_TYPE_WIFI)
 			return __connman_error_not_supported(msg);

-		if (strlen(str) < 8 || strlen(str) > 63)
+		if (strlen(str) > 63)
 			return __connman_error_passphrase_required(msg);

 		if (g_strcmp0(technology->tethering_passphrase, str) != 0) {
