--- Python-3.3.0/Makefile.pre.in.orig	2014-02-04 15:12:27.938317209 -0500
+++ Python-3.3.0/Makefile.pre.in	2014-02-04 15:29:22.994346615 -0500
@@ -35,7 +35,8 @@
 AR=		@AR@
 RANLIB=		@RANLIB@
 READELF=	@READELF@
-SOABI=		@SOABI@
+# This is a terrible hack to get around an issues with the 'dm' ABIFLAG
+SOABI=		cpython-33m
 LDVERSION=	@LDVERSION@
 HGVERSION=	@HGVERSION@
 HGTAG=		@HGTAG@
@@ -109,7 +110,9 @@
 INCLUDEDIR=	@includedir@
 CONFINCLUDEDIR=	$(exec_prefix)/include
 SCRIPTDIR=	$(prefix)/lib
-ABIFLAGS=	@ABIFLAGS@
+# This is a terrible hack to get around an issue with the ABIFLAGS being set to 'dm'
+# for debug builds
+ABIFLAGS=	m
 
 # Detailed destination directories
 BINLIBDEST=	$(LIBDIR)/python$(VERSION)
@@ -118,7 +121,9 @@
 CONFINCLUDEPY=	$(CONFINCLUDEDIR)/python$(LDVERSION)
 
 # Symbols used for using shared libraries
-SO=		@SO@
+# This is a terrible hack to get around an issue with the ABIFLAGS being set to 'dm'
+# for debug builds
+SO=		.cpython-33m.so
 LDSHARED=	@LDSHARED@ $(PY_LDFLAGS)
 BLDSHARED=	@BLDSHARED@ $(PY_LDFLAGS)
 LDCXXSHARED=	@LDCXXSHARED@
