--- python3-3.3.0.orig/Modules/_ssl.c	2012-09-29 04:00:44.000000000 -0400
+++ python3-3.3.0/Modules/_ssl.c	2014-12-08 12:23:09.796826042 -0500
@@ -1101,7 +1101,9 @@
     PyObject *version;
     PyObject *sn_obj;
     PyObject *obj;
+    PyObject *pubkey_obj;
     ASN1_INTEGER *serialNumber;
+    const ASN1_BIT_STRING *pubkey;
     char buf[2048];
     int len;
     ASN1_TIME *notBefore, *notAfter;
@@ -1140,6 +1142,16 @@
     }
     Py_DECREF(version);
 
+    pubkey = X509_get0_pubkey_bitstr(certificate);
+    pubkey_obj = PyBytes_FromStringAndSize(pubkey->data, pubkey->length);
+    if (pubkey_obj == NULL)
+        goto fail0;
+    if (PyDict_SetItemString(retval, (const char *)"pubkey", pubkey_obj) < 0) {
+        Py_DECREF(pubkey_obj);
+        goto fail0;
+    }
+    Py_DECREF(pubkey_obj);
+
     /* get a memory buffer */
     biobuf = BIO_new(BIO_s_mem());
 
